### PR TITLE
[CI] Switching to micromamba

### PR DESCRIPTION
**Pull request by tdegeus**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -69,7 +57,7 @@ jobs:
     - name: Configure using CMake (Windows)
       if: runner.os == 'Windows'
       run: |
-        conda install -c conda-forge clang_win-64 ninja
+        mamba install -c conda-forge clang_win-64 ninja
```
Comment (wolfv) [inline .github/workflows/ci.yml:60]: micromamba ...

Comment (tdegeus) [reply .github/workflows/ci.yml:60]: Thanks!

**Comment by wolfv** [comment]
@tdegeus there is also a new secret trick for environment files:

you can use

```
- mydep
- sel(win): clang
- sel(unix): gcc
```

to use selectors and install certain dependencies only when some requirement is true. Only very simple conditions work currently (linux, unix, osx and win).

**Comment by tdegeus** [comment]
That is very nice @wolfv ! Is is mamba that does it, or is it also already integrated in conda?

**Comment by tdegeus** [comment]
@wolfv Is there a way to install from more than one environment variable? Suppose that I want to select the compiler just in the CI, and that I want to keep a more general enviroment.yaml that a user could use directly (even when not using mamba).

**Comment by wolfv** [comment]
Not yet. Could be interesting to think about it.

**Pull request merged**